tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -867,16 +857,41 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
       }
       return;
     };
+
+    // Get the possible padded shape.
+    auto pad_shape = [&batch_dims](const Shape old_shape) {
```

Comment:
Use `pad_shape` it in `pad_operand` to avoid the duplicated code which calls `RoundUpTo`.

Also it might be worth making `pad_shape` and `pad_operand` top-level functions instead of lambda functions.